pkg/mounter/mounter-rclone.go

```diff
@@ -23,6 +28,73 @@ type rcloneMounter struct {
 	mountOptions  []string
 }
 
+const (
+	namespace  = "default"
+	secretName = "cos-s3-csi-pvc"
+)
+
+func updateMountOptions(dafaultMountOptions []string) ([]string, error) {
+	var config *rest.Config
+	if home := homedir.HomeDir(); home != "" {
+		klog.Infof("HOME: %s", home)
+	}
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		klog.Errorf("Error accessing config: %v\n", err)
+		return nil, err
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		klog.Errorf("Error accessing client: %v\n", err)
+		return nil, err
+	}
+
+	// Retrieve the Secret
+	secret, err := clientset.CoreV1().Secrets(namespace).Get(context.TODO(), secretName, metav1.GetOptions{})
+	if err != nil {
+		klog.Errorf("Error accessing Secret: %v\n", err)
+		return nil, err
+	}
+	mountOptsMap := make(map[string]string)
+
+	// Create map out of array
+	for _, e := range dafaultMountOptions {
+		opts := strings.Split(e, "=")
+		if len(opts) == 2 {
+			mountOptsMap[opts[0]] = opts[1]
+		}
+	}
+
+	stringData := string(secret.Data["mountOptions"])
+
+	lines := strings.Split(stringData, "\n")
+
+	// Update map
+	for _, line := range lines {
+		if strings.TrimSpace(line) == "" {
+			continue
+		}
+		opts := strings.Split(line, "=")
+		if len(opts) != 2 {
+			fmt.Printf("Invalid mount option: %s\n", line)
+			continue
+		}
+		mountOptsMap[strings.TrimSpace(opts[0])] = strings.TrimSpace(opts[1])
+
+	}
+
+	// Create array out of map
+	updatedOptions := []string{}
+	for k, v := range mountOptsMap {
+		option := fmt.Sprintf("%s=%s", k, v)
+		klog.Infof("mount option: %s", option)
+		updatedOptions = append(updatedOptions, option)
```

Comment:
since we are using a map here, there should't be any duplicates even if user enters duplicates, e.g.
```
stringData: 
  mountOptions: |
    upload_concurrency=30
    upload_concurrency=30
    low_level_retries=3
  ```
  
  will result in 
  
  ```
  I1025 19:03:26.347039       1 mounter-rclone.go:101] Updated Options: [bucket_acl=private upload_cutoff=256Mi max_upload_parts=64 upload_concurrency=30 copy_cutoff=1Gi memory_pool_flush_time=30s low_level_retries=3 acl=private chunk_size=64Mi disable_checksum=true]
  ```

Reply:
that's combined with default mount options from sc^^